client/src/api/block_builder/input_selection/new/requirement/sender.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{InputSelection, Requirement};
+use crate::{
+    block::address::Address,
+    error::{Error, Result},
+    secret::types::InputSigningData,
+};
+
+fn has_ed25519_address(input: &InputSigningData, address: &Address) -> bool {
+    // TODO could also be in state/governor?
+    if let Some(unlock_conditions) = input.output.unlock_conditions() {
+        if let Some(address_unlock_condition) = unlock_conditions.address() {
+            address_unlock_condition.address() == address
```

Comment:
Shouldn't we check the expiration unlock condition before? Because if it's expired, then the address unlock condition will not be used

Reply:
It'll be done once at the beginning, but it's not done yet, yes

Reply:
But even if you check it at the beginning, you still want to include expired and not expired if your address is in the correct unlock condition?